app/services/web_mentions/web_mention_handler.rb

```diff
@@ -0,0 +1,44 @@
+module WebMentions
+  class WebMentionHandler
+    def initialize(canonical_url, article_url = nil, webmention_url = nil)
```

Comment:
can we use keyword parameters? switching the two optional URLs is bound to happen :D